packages/frontend/src/components/ControlledAutocomplete/AddNewOptionModal.tsx

```diff
@@ -0,0 +1,125 @@
+import { FormEvent, useCallback, useState } from 'react'
+import { useMutation } from '@apollo/client'
+import {
+  FormControl,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalHeader,
+  ModalOverlay,
+} from '@chakra-ui/react'
+import { Button, FormLabel, Input } from '@opengovsg/design-system-react'
+import client from 'graphql/client'
+import { CREATE_TABLE } from 'graphql/mutations/tiles/create-table'
+import { GET_DYNAMIC_DATA } from 'graphql/queries/get-dynamic-data'
+
+interface AddNewOptionalModalProps {
+  modalHeader: string
+  onSubmit: (newValue: string) => void
+  onClose: () => void
```

Comment:
nit: `onSubmitError` callback would be nice

Reply:
wont fail now since it doesnt call any mutation